metabase/metabase
# Dashboard sections

Closes #38209 
Fixes: #35667

Adds a concept of "sections" and "empty cards" to dashboards:

- a placeholder card is just a blank card, but can be replaced with a question using the new "replace question" action (#36744). Technically they're implemented as virtual cards with `display: "placeholder"` and their own viz component (`DashCardPlaceholder`)
- sections are pre-defined common layouts that might help people to focus on the dashboard content vs. layout fiddling. They use placeholder cards (although not limited to only them)

### To verify

1. Open a dashboard and enter the edit mode
2. Click the "Add section" button at the top right
3. Pick one of the available sections
4. Ensure it's adding a few cards to the end of the current dashboard tab
5. Click "Select question" on a placeholder card to replace it with a question
6. Ensure changes get persisted (also parameters, viz setting overrides, etc.)

Note: it's expected that placeholder cards show up as blank outside the editing mode.

### Demo

https://github.com/metabase/metabase/assets/17258145/56870e87-fedb-4765-b07c-2b976c0c9ca7
